packages/zowe-explorer/src/utils/ReleaseNotes.ts

```diff
@@ -0,0 +1,182 @@
+        const releaseNotes = await this.getReleaseNotes();
+        const changelog = await this.getChangelog();
+        const displayReleaseNotesSetting = SettingsConfig.getDirectValue(Constants.SETTINGS_DISPLAY_RELEASE_NOTES);
+        const versionOptions = await this.getAllMajorMinorVersions();
+
+        return this.panel.webview.postMessage({
+            releaseNotes: releaseNotes,
+            changelog: changelog,
+            version: this.version,
+            displayReleaseNotesSetting: displayReleaseNotesSetting,
+            dropdownOptions: Constants.RELEASE_NOTES_OPTS_LABELS,
+            versionOptions: versionOptions,
+        });
+    }
+
+    public async getReleaseNotes(): Promise<string> {
+        const releaseNotesPath = this.context.asAbsolutePath(`src/webviews/dist/resources/release-notes.md`);
+        try {
+            const releaseNotes = await fs.readFile(releaseNotesPath, { encoding: "utf8" });
+            return this.extractCurrentVersionNotes(releaseNotes, "release notes");
+        } catch (error) {
+            ZoweLogger.error(`[ReleaseNotes] Error reading release notes file: ${String(error)}`);
+            return l10n.t("No release notes found for this version.");
+        }
+    }
+
+    public async getChangelog(): Promise<string> {
+        const changelogPath = this.context.asAbsolutePath("CHANGELOG.md");
+        try {
+            const changelog = await fs.readFile(changelogPath, { encoding: "utf8" });
+            return this.extractCurrentVersionNotes(changelog, "changelog");
+        } catch (error) {
+            ZoweLogger.error(`[ReleaseNotes] Error reading changelog file: ${String(error)}`);
+            return l10n.t("No changelog entries found for this version.");
+        }
+    }
+
+    public extractCurrentVersionNotes(changelog: string, type: string): string {
+        const regex = new RegExp(`## \`(${this.version}\\.\\d+)\`[\\s\\S]*?(?=^## \`(?!${this.version}\\.\\d+))`, "gm");
+        const matches = [...changelog.matchAll(regex)];
+        if (matches.length === 0) {
+            return l10n.t("No {0} entries found for this version.", type);
+        }
+
+        return matches.map((m) => m[0].trim()).join("\n\n");
+    }
+
+    public async getAllMajorMinorVersions(): Promise<string[]> {
+        const changelogPath = this.context.asAbsolutePath("CHANGELOG.md");
+        try {
+            const changelog = await fs.readFile(changelogPath, { encoding: "utf8" });
+            // Match all ## `x.y.z` and extract x.y
+            const regex = /^## `(\d+\.\d+)\.\d+`/gm;
+            const versions = new Set<string>();
+            let match;
+            while ((match = regex.exec(changelog)) !== null) {
+                versions.add(match[1]);
+            }
+            if (!versions.has(this.version)) {
+                versions.add(this.version);
+            }
+            if (!versions.has(this.extensionVersion)) {
+                versions.add(this.extensionVersion);
```

Comment:
(nitpick) Since it's a set, you don't need to check if it contains the value as its elements are de-duplicated